### DISCUSSION TITLE
Autocompletion not working

### DISCUSSION THREAD

**Discussion by aamasse16** [zed-industries/zed#6494] [Feature Requests]
I am having an issue with my auto complete, its simply not there at all. This could just be something I'm overlooking or something like that. I am currently developing in typescript and thought maybe there isn't autocompletion support for ts. I also have tired this with JavaScript and its still not working so maybe someone else has had this problem and found a solution?

**Comment by JosephTLyons** [comment]
Hey @aamasse16, are you able to create a simple TypeScript / JavaScript project that replicates the issue?  I'd like to open some code you are having an issue with on my end and see if I run into it as well.

**Reply by aamasse16** [reply]
okay i just started to make one and after i ran "touch test.ts" to make a typescript file i got this error
Language server error: TypeScript

failed to run npm info

Caused by:
    No such file or directory (os error 2)

didnt see this before but im sure this is the root of the problem. i do have node installed and npm is operational so im unsure why this is happening

**Comment by aamasse16** [comment]
I was able to fix this buy uninstalling zed and reinstalling it, not sure why this worked but it did and I'm not going to complain.

**Reply by JosephTLyons** [reply]
Strange.  Sorry for the poor experience here, but glad it is working now.

**Reply by Dandandans** [reply]
Same issue. Uninstalling and reinstalling did not fix it. 

```Language server error: TSX

failed to execute npm info:
stdout: "\nUsage: npm <command>\n\nwhere <command> is one of:\n    access, adduser, audit, bin, bugs, c, cache, ci, cit,\n    clean-install, clean-install-test, completion, config,\n    create, ddp, dedupe, deprecate, dist-tag, docs, doctor,\n    edit, explore, fund, get, help, help-search, hook, i, init,\n    install, install-ci-test, install-test, it, link, list, ln,\n    login, logout, ls, org, outdated, owner, pack, ping, prefix,\n    profile, prune, publish, rb, rebuild, repo, restart, root,\n    run, run-script, s, se, search, set, shrinkwrap, star,\n    stars, start, stop, t, team, test, token, tst, un,\n    uninstall, unpublish, unstar, up, update, v, version, view,\n    whoami\n\nnpm <command> -h  quick help on <command>\nnpm -l            display full usage info\nnpm help <term>   search for help on <term>\nnpm help npm      involved overview\n\nSpecify configs in the ini-formatted file:\n    /Users/dab/.npmrc\nor on the command line via: npm <command> --key value\nConfig info can be viewed via: npm help config\n\nnpm@6.14.17 /usr/local/lib/node_modules/npm\n\n"
stderr: ""```

**Reply by aamasse16** [reply]
hey what node version are you using?

**Reply by Dandandans** [reply]
`v14.21.1`

**Reply by aamasse16** [reply]
okay im not the expert here but these are my versions and im not sure why you are getting that error.
  npm: '9.5.0',
  node: '18.15.0',
im assuming that you have been using another code editor and its been working?

**Reply by Dandandans** [reply]
I should add I'm using `n` to manage node.

**Reply by aamasse16** [reply]
ive never used `n` however i have used nvm. maybe its a pathing issue? are you using the same terminal in Zed as you have in other editors?

**Reply by Dandandans** [reply]
Like @christopherferreira3, I installed node 18.15.0, and it now works.

**Comment by ghost** [comment]
Hey hey! I'm actually bumping into the same error. 
Not sure if it helps but I'm using an M1 Macbook Pro and I'm trying with a .ts file simply containing a console.log with hello world.

**Reply by aamasse16** [reply]
have you tried uninstalling and reinstalling? also what is the error you are getting if any?

**Reply by ghost** [reply]
Yeah I did. This is the error I'm getting:

"Language server error: TypeScript

failed to run npm info

Caused by:
No such file or directory (os error 2)"

I'm using node with nvm and I have multiple versions installed.

**Reply by ghost** [reply]
Managed to make it work with 

`nvm alias default 18.X.X`

In my specific case I didn't have node v18 so I had to install it and revert back the default option for the one I use.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
